wp_api_integration_tests/tests/test_plugins_mut.rs

```diff
@@ -1,32 +1,29 @@
 use rstest::rstest;
 use serial_test::serial;
 use wp_api::plugins::{PluginCreateParams, PluginSlug, PluginStatus, PluginUpdateParams};
-use wp_api_integration_tests::wp_db;
+use wp_api_integration_tests::backend::RestoreServer;
 use wp_api_integration_tests::{
-    api_client, run_and_restore_wp_content_plugins, AssertResponse, CLASSIC_EDITOR_PLUGIN_SLUG,
-    HELLO_DOLLY_PLUGIN_SLUG, WP_ORG_PLUGIN_SLUG_CLASSIC_WIDGETS,
+    api_client, AssertResponse, CLASSIC_EDITOR_PLUGIN_SLUG, HELLO_DOLLY_PLUGIN_SLUG,
+    WP_ORG_PLUGIN_SLUG_CLASSIC_WIDGETS,
 };
 
 #[tokio::test]
 #[serial]
 async fn create_plugin() {
-    run_and_restore_wp_content_plugins(|| {
-        wp_db::run_and_restore(|mut _db| async move {
-            let status = PluginStatus::Active;
-            let params = PluginCreateParams {
-                slug: WP_ORG_PLUGIN_SLUG_CLASSIC_WIDGETS.into(),
-                status,
-            };
-            let created_plugin = api_client()
-                .plugins()
-                .create(&params)
-                .await
-                .assert_response();
-            assert_eq!(created_plugin.status, status);
-            println!("Created Plugin: {:?}", created_plugin);
-        })
-    })
-    .await;
+    let status = PluginStatus::Active;
+    let params = PluginCreateParams {
+        slug: WP_ORG_PLUGIN_SLUG_CLASSIC_WIDGETS.into(),
+        status,
+    };
+    let created_plugin = api_client()
+        .plugins()
+        .create(&params)
+        .await
+        .assert_response();
+    assert_eq!(created_plugin.status, status);
+    println!("Created Plugin: {:?}", created_plugin);
+
+    RestoreServer::all().await;
```

Comment:
Does this line run if any assertion fails? If it doesn't, is there any way to make it always run even if this test function fails, so that its failure won't affect other tests?

Reply:
It doesn't at the moment. It's possible to always restore, but I am not sure about the best approach for that within the current implementation. So, I'll leave that to a future improvement.

I _think_ I might be able to address this as part of my work for adding function attributes macros such as `#[mutable_test]`. I intend to handle the restoration within that macro, so it's a good chance to address this as well.

Thank you for bringing it up!